src/mem/mod.rs

```diff
@@ -9,3 +9,25 @@ mod arena;
 pub use arena::*;
 
 pub mod cow;
+
+/// Aligns the given address to the alignment for the given type.
+///
+/// `align` must be a power of two; otherwise, the returned value
+/// will be well-defined but unspecified.
+///
+/// This function will always return a value greather than or equal to `addr`.
+#[inline]
+pub(in crate) fn align_to(addr: usize, align: usize) -> usize {
+    let mask = align.saturating_sub(1);
```

Comment:
I assume this is a saturating subtraction so that if `align` is 0 it won't panic?

It seems to me that using `wrapping_sub` would generate smaller code, and still meet the description "well-defined but unspecified" in the documentation.